assets/src/hooks/useLocationSearchResults.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "react"
+import { fetchLocationSearchResults } from "../api"
+import { LocationSearchResult } from "../models/locationSearchResult"
+
+export const useLocationSearchResults = (
+  text: string | null
+): LocationSearchResult[] | null => {
+  const [searchResults, setSearchResults] = useState<
+    LocationSearchResult[] | null
+  >(null)
+
+  useEffect(() => {
+    let shouldUpdate = true
```

Comment:
praise: Nice work considering the race condition! I often forget this - wonder if it is needed for the vehicle result hooks?

Reply:
I think for the vehicle results the current approach is fine - `useChannel` and its derivatives call `channel.leave()` in the cleanup function of their `useEffect` hooks, so presumably once that has been called the callback added to that channel by calling `channel.on()` will no longer be invoked. With HTTP API requests it's different because the various `fetch*` functions return a promise that could resolve after the parameters of the hook have changed.